api/api.go

```diff
@@ -2,48 +2,63 @@ package api
 
 import (
 	"encoding/json"
-	"errors"
 	"fmt"
 	"net/http"
 	"strings"
 
+	"github.com/gorilla/mux"
 	"github.com/soundcloud/periskop/metrics"
 	"github.com/soundcloud/periskop/repository"
 )
 
-func NewHandler(r *repository.ErrorsRepository) http.Handler {
+func NewServicesListHandler(r *repository.ErrorsRepository) http.Handler {
 	return http.HandlerFunc(func(w http.ResponseWriter, req *http.Request) {
-		// Allow CORS requests for local development since API and frontend run on different ports
-		origin := req.Header.Get("Origin")
-		if strings.HasPrefix(origin, "http://localhost:") {
-			w.Header().Set("Access-Control-Allow-Origin", origin)
+		allowCORS(w, req)
+		err := servicesList(w, r)
+		if err != nil {
+			metrics.ErrorCollector.ReportWithHTTPRequest(err, req)
 		}
+	})
+}
 
-		path := strings.TrimPrefix(req.URL.Path, "/services/")
+func NewErrorsListHandler(r *repository.ErrorsRepository) http.Handler {
+	return http.HandlerFunc(func(w http.ResponseWriter, req *http.Request) {
+		allowCORS(w, req)
+		vars := mux.Vars(req)
 		numberOfOccurrencesPerError := 10
 
-		if len(path) == 0 {
-			err := servicesList(w, r)
-			if err != nil {
-				metrics.ErrorCollector.ReportWithHTTPRequest(err, req)
-			}
-		} else if service, err := extractServiceName(path); err == nil {
-			err = errorsForService(w, r, service, numberOfOccurrencesPerError)
+		if service, found := vars["service_name"]; found {
+			err := errorsForService(w, r, service, numberOfOccurrencesPerError)
 			if err != nil {
 				metrics.ErrorCollector.ReportWithHTTPRequest(err, req)
 			}
 		} else {
-			metrics.ErrorCollector.ReportWithHTTPRequest(err, req)
 			http.NotFound(w, req)
 		}
 	})
 }
 
-func extractServiceName(url string) (string, error) {
-	if strings.HasSuffix(url, "/errors/") {
-		return strings.TrimSuffix(url, "/errors/"), nil
+func NewErrorDeleteHandler(r *repository.ErrorsRepository) http.Handler {
+	return http.HandlerFunc(func(w http.ResponseWriter, req *http.Request) {
+		allowCORS(w, req)
```

Comment:
nit: can you configure this with a filter so that we don't have to do it in all the handlers?

Reply:
we can use [this](https://github.com/gorilla/mux#handling-cors-requests) but it's gonna enable `Access-Control-Allow-Origin` for any host and not only `localhost`.

Reply:
I guess we could create our custom [middleware](https://github.com/gorilla/mux#middleware) that sets the header only for `localhost`

Reply:
Ok thanks, i'll do it! :+1: